data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/impl/ReceiverVerticle.java

```diff
@@ -0,0 +1,117 @@
+import io.vertx.core.http.HttpServerRequest;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.function.Function;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static dev.knative.eventing.kafka.broker.core.utils.Logging.keyValue;
+import static io.netty.handler.codec.http.HttpResponseStatus.NOT_FOUND;
+
+public class ReceiverVerticle extends AbstractVerticle {
+
+  private static final Logger logger = LoggerFactory.getLogger(ReceiverVerticle.class);
+
+  private final HttpServerOptions httpServerOptions;
+  private final Function<Vertx, IngressProducerReconcilableStore> ingressProducerStoreFactory;
+  private final Iterable<Handler<HttpServerRequest>> preHandlers;
+  private final IngressRequestHandler ingressRequestHandler;
+
+  private HttpServer server;
+  private MessageConsumer<Object> messageConsumer;
+  private IngressProducerReconcilableStore ingressProducerStore;
+
+  public ReceiverVerticle(HttpServerOptions httpServerOptions,
+                          Function<Vertx, IngressProducerReconcilableStore> ingressProducerStoreFactory,
+                          Iterable<Handler<HttpServerRequest>> preHandlers,
+                          IngressRequestHandler ingressRequestHandler) {
+    Objects.requireNonNull(ingressProducerStoreFactory);
+    Objects.requireNonNull(ingressRequestHandler);
+
+    this.httpServerOptions = httpServerOptions != null ? httpServerOptions : new HttpServerOptions();
+    this.ingressProducerStoreFactory = ingressProducerStoreFactory;
+    this.preHandlers = preHandlers != null ? preHandlers : Collections.emptyList();
+    this.ingressRequestHandler = ingressRequestHandler;
+  }
+
+
+  @Override
+  public void start(final Promise<Void> startPromise) {
+    this.ingressProducerStore = this.ingressProducerStoreFactory.apply(vertx);
+    this.messageConsumer = ResourcesReconciler
+      .builder()
+      .watchIngress(this.ingressProducerStore)
+      .buildAndListen(vertx);
+
+    this.server = vertx.createHttpServer(httpServerOptions);
+
+    this.server.requestHandler(this::handle)
+      .exceptionHandler(startPromise::tryFail)
+      .listen(httpServerOptions.getPort(), httpServerOptions.getHost())
+      .<Void>mapEmpty()
+      .onComplete(startPromise);
+  }
+
+  @Override
+  public void stop(Promise<Void> stopPromise) {
+    CompositeFuture.all(
+      server.close().mapEmpty(),
+      messageConsumer.unregister()
+    )
+      .<Void>mapEmpty()
+      .onComplete(stopPromise);
+  }
+
+  private void handle(HttpServerRequest request) {
+    // Run preHandlers
+    for (Handler<HttpServerRequest> preHandler : this.preHandlers) {
+      preHandler.handle(request);
+      // preHandler might end the request prematurely
+      if (request.isEnded()) {
+        return;
+      }
+    }
+
+    // Look up for the ingress producer
+    IngressProducer producer = this.ingressProducerStore.resolve(request.path());
+    if (producer == null) {
```

Comment:
when would we hit this?

Reply:
if you're doing a request to a broker path that does not exists